src/createXcmTypes/util/sortMultiAssetsAscending.spec.ts

```diff
@@ -8,208 +9,1327 @@ describe('sortMultiAssetsAscending', () => {
+				fun: {
+					Fungible: '100000',
+				},
+				id: {
+					Concrete: mockSystemApi.registry.createType('MultiLocation', {
+						parents: 1,
+						interior: mockSystemApi.registry.createType(
+							'InteriorMultiLocation',
+							{
+								X2: [
+									{ Parachain: '2000' },
+									{ GeneralKey: '0xA73397cE0cCFdE92e7B23F3d0C462eF099E9E978' },
+								],
+							}
+						),
+					}),
+				},
+			},
+			{
+				fun: {
+					Fungible: '100000',
+				},
+				id: {
+					Concrete: mockSystemApi.registry.createType('MultiLocation', {
+						parents: 1,
+						interior: mockSystemApi.registry.createType(
+							'InteriorMultiLocation',
+							{
+								X2: [
+									{ Parachain: '2023' },
+									{ GeneralKey: '0xA73397cE0cCFdE92e7B23F3d0C462eF099E9E978' },
+								],
+							}
+						),
+					}),
+				},
+			},
+		];
+
+		const res = sortMultiAssetsAscending(multiAssets);
+
+		expect(res.length).toEqual(expected.length);
+		expect(res[0].id).toEqual(expected[0].id);
+		expect(res[res.length - 1].fun).toEqual(expected[expected.length - 1].fun);
+		expect(JSON.stringify(res)).toEqual(JSON.stringify(expected));
+	});
+
+	it('Should correctly sort based on differing third Junction key value for X3s when all other fields are the same', () => {
+		const multiAssets: MultiAsset[] = [
+			{
+				fun: {
+					Fungible: '200000000',
+				},
+				id: {
+					Concrete: mockSystemApi.registry.createType('MultiLocation', {
+						parents: 1,
+						interior: mockSystemApi.registry.createType(
+							'InteriorMultiLocation',
+							{
+								X3: [
+									{ Parachain: '2000' },
+									{ PalletInstance: '50' },
+									{ GeneralIndex: '1984' },
+								],
+							}
+						),
+					}),
+				},
+			},
+			{
+				fun: {
+					Fungible: '200000000',
+				},
+				id: {
+					Concrete: mockSystemApi.registry.createType('MultiLocation', {
+						parents: 1,
+						interior: mockSystemApi.registry.createType(
+							'InteriorMultiLocation',
+							{
+								X3: [
+									{ Parachain: '2000' },
+									{ PalletInstance: '50' },
+									{ GeneralIndex: '1' },
+								],
+							}
+						),
+					}),
+				},
+			},
+		];
+
+		const expected: MultiAsset[] = [
+			{
+				fun: {
+					Fungible: '200000000',
+				},
+				id: {
+					Concrete: mockSystemApi.registry.createType('MultiLocation', {
+						parents: 1,
+						interior: mockSystemApi.registry.createType(
+							'InteriorMultiLocation',
+							{
+								X3: [
+									{ Parachain: '2000' },
+									{ PalletInstance: '50' },
+									{ GeneralIndex: '1' },
+								],
+							}
+						),
+					}),
+				},
+			},
+			{
+				fun: {
+					Fungible: '200000000',
+				},
+				id: {
+					Concrete: mockSystemApi.registry.createType('MultiLocation', {
+						parents: 1,
+						interior: mockSystemApi.registry.createType(
+							'InteriorMultiLocation',
+							{
+								X3: [
+									{ Parachain: '2000' },
+									{ PalletInstance: '50' },
+									{ GeneralIndex: '1984' },
```

Comment:
what happens in this scenario when asset 1 and asset 3 are the same? will the API just construct a call with two asset repeated? I think this is not permitted by the XCM format, the asset amounts should be added and placed in only one array element.

Reply:
great catch! right now the api is checking for dupes at the `MultiAsset` level and not the assetId/MultiLocation level. Will update this to throw an error when duplicate `assetId`/ `MultiLocations` are passed in.